### PR TITLE
Jvenstad/remove old job status

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review. 

The three first commits are part of an existing PR. The next three commits are also intended as a separate PR, before merging the last commit here.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java [hunk 1]
```diff
@@ -124,11 +119,6 @@
 
     // JobStatus field
     private static final String jobTypeField = "jobType";
-    private static final String errorField = "jobError";
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java]: Does the previous version handle removal of these fields?

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java [hunk 2]
```diff
@@ -336,6 +336,12 @@ public void testApplicationApi() {
 
         deploymentTester.triggerJobs();
 
+        // POST a triggering to force a production job to start without successful tests
+        tester.assertResponse(request("/application/v4/tenant/tenant2/application/application2/instance/instance1/job/production-us-west-1", POST)
+                                      .data("{\"skipTests\":true}")
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java]: 😆

**Review by mpolden — COMMENTED** [review]
Nice cleanup!

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java]: Yes, all fields were already optional, except the job type.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java]: 🙈

**Comment by jonmv** [comment]
Merging it all together. Loss of the old job status data shouldn't be a problem, as the new one is already used for everything important.

**Pull request merged**